src/Factory.php

```diff
@@ -52,6 +53,10 @@ public function __construct(string $class, $defaultAttributes = [])
 
         $this->class = $class;
         $this->attributeSet[] = $defaultAttributes;
+
+        if ($this->classIsEmbeddedObject()) {
+            $this->persist = false;
+        }
```

Comment:
should be in `Factory` constructor in order to be applied to `AnonymousClass` as well

Reply:
Can we not move this to `Factory::isPersisting()`?